data/data/azure/vnet/vnet.tf

```diff
@@ -17,6 +17,7 @@ resource "azurerm_subnet" "master_subnet" {
   ] : cidr.value if cidr.include]
   virtual_network_name = local.virtual_network
   name                 = var.azure_control_plane_subnet
+  service_endpoints    = ["Microsoft.Storage"]
```

Comment:
Hm... why do we need this? What is it doing?

Reply:
We need that to link the subnet to the storage account. 
If we don't have it we get 
ERROR Error: retrieving Storage Account: (Name "*" / Resource Group "*-rg"): storage.AccountsClient#GetProperties: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="StorageAccountNotFound" Message="The storage account clusterocksi was not found."